experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,666 @@
+        return G
+    end
+end
+
+function Base.show(io::IO, G::ReductiveGroup)
+    io = AbstractAlgebra.pretty(io)
+    if G.group[1] == :SL
+        print(io, "Reductive group ", G.group[1], G.group[2])
+    else
+        println(io, "Torus of degree ", group(G)[2])
+        print(IOContext(io, :supercompact => true), AbstractAlgebra.Indent(), "over the field ", field(G))
+        print(io, AbstractAlgebra.Dedent())
+    end
+end
+
+reductive_group(sym::Symbol, m::Int, R::MPolyRing) = ReductiveGroup(sym,m,R)
+reductive_group(sym::Symbol, m::Int, F::Field) =  ReductiveGroup(sym,m,F)
+group(G::ReductiveGroup) = G.group
+field(G::ReductiveGroup) = G.field
+reynolds_operator(G::ReductiveGroup) = G.reynolds_operator
+group_ideal(G::ReductiveGroup) = G.group_ideal
+canonical_representation(G::ReductiveGroup) = G.canonical_representation
+natural_representation(G::ReductiveGroup) = G.canonical_representation
+
+#####################
+#Representation of Reductive Groups. Embeds SLm in GLn. 
+#####################
+
+mutable struct RepresentationReductiveGroup
+    group::ReductiveGroup
+    rep_mat::AbstractAlgebra.Generic.MatSpaceElem
+    sym_deg::Tuple{Bool, Int}
+    reynolds_v::Function
+
+    weights::Vector{Vector{ZZRingElem}}
+    
+    #representation of group G over symmetric degree d
+    function RepresentationReductiveGroup(G::ReductiveGroup, d::Int)
+        R = new()
+        R.group = G
+        R.rep_mat = rep_mat_(G, d)
+        R.sym_deg = (true, d)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+
+    #function RepresentationReductiveGroup(G::ReductiveGroup, W::Union{ZZMatrix, Matrix{<:Integer}, Vector{<:Integer}})
+    function RepresentationReductiveGroup(G::ReductiveGroup, W::Vector{Vector{ZZRingElem}})
+        if !(G.group[1] == :torus)
+            return nothing
+        else
+            R = new()
+            R.group = G
+            R.weights = W
+            return R
+        end
+    end
+    
+    #matrix M is the representation matrix. does not check M.
+    function RepresentationReductiveGroup(G::ReductiveGroup, M::AbstractAlgebra.Generic.MatSpaceElem)
+        @req base_ring(M) == base_ring(G.group_ideal) "Group ideal and representation matrix must have same parent ring"
+        R = new()
+        R.group = G
+        R.rep_mat = M
+        R.sym_deg = (false, 0)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+end
+
+#representation_reductive_group(G::ReductiveGroup, d::Int) = RepresentationReductiveGroup(G, d)
+representation_reductive_group(G::ReductiveGroup, M::AbstractAlgebra.Generic.MatSpaceElem) = RepresentationReductiveGroup(G,M)
+group(R::RepresentationReductiveGroup) = R.group
+
+function representation_on_forms(G::ReductiveGroup, d::Int)
+    if G.group[1] == :SL
+        return RepresentationReductiveGroup(G, d)
+    end
+end
+
+function representation_on_weights(G::ReductiveGroup, W::Union{ZZMatrix, Matrix{<:Integer}, Vector{<:Int}})
```

Comment:
Maybe this should be called `representation_from_weights`? Not really sure. @wdecker

Reply:
I agree that `representation_on_weights` is a bit misleading, and should be avoided. The given alternatives both sound OK to me. I have no strong preference either way (and if someone comes up with an even better name in the future, we can always rename, and keep the old name as an alias to ensure backwards compatibility). Another variant would be `representation_with_weights`.

If you want me to make a decision, I'll pick `representation_from_weights` but really I am happy with either of them.